include/simdjson/generic/numberparsing.h

```diff
@@ -8,7 +8,7 @@ namespace ondemand {
 /**
  * The type of a JSON number
  */
-enum class number_type {
+enum class number_type : uint8_t {
```

Comment:
small thing i noticed, good to codify this clearly because we treat it as a uint8_t already